test/e2e/network/endpointslice.go

```diff
@@ -730,6 +731,141 @@ var _ = common.SIGDescribe("EndpointSlice", func() {
 
 	})
 
+	ginkgo.It("should delete EndpointSlices and create mirrored EndpointSlices when Service selector is made to empty", func(ctx context.Context) {
```

Comment:
this looks good

Reply:
what happens now if I remove the selector and the I add it again?

Reply:
yes, enhanced the test now to add selector back into the service.

Reply:
it was more a curious question, what happen with the mirrored slice?

Reply:
@aojea Not sure I fully understand your question. Here is what happens with and without fix. Hope it helps.

Without fix:
1. Upon service creation for a pod, it creates endpoint and endpointslice object managed by endpointslice controller.
2. Update service without selector, then it retains objects created in step 1 and endpointslice mirror object is created by endpointslicemirroring controller.
3. Update endpoint object (for example, port number of the endpoint), then it updates only endpointslice mirror object but not endpointslice object managed by endpointslice controller.
4. Add selector label back into service, then it deletes endpointslice mirror object and endpoint object is updated back with original port number defined in the service.

With fix:
1. Upon service creation for a pod, it creates endpoint and endpointslice object managed by endpointslice controller.
2. Update service without selector, then it deletes endpointslice object created in step 1 and endpointslice mirror object is created by endpointslicemirroring controller.
3. Update endpoint object (for example, port number of the endpoint), then it updates only endpointslice mirror object.
4. When selector label is added back into service, then it deletes endpointslice mirror object, endpoint object is updated with original port number defined in the service and adds endpointslice object which is managed by endpointslice controller.